modules/leaflet/src/deck-layer.ts

```diff
@@ -0,0 +1,208 @@
+// deck.gl-community
+// SPDX-License-Identifier: MIT
+// Copyright (c) vis.gl contributors
+
+import * as L from 'leaflet';
+import type {Deck, DeckProps} from '@deck.gl/core';
+import {createDeckInstance, updateDeckView} from './deck-utils';
+
+export class DeckLayer extends L.Layer {
+  props: DeckProps;
+  _container: HTMLDivElement | undefined = undefined;
+  _deck: Deck | undefined = undefined;
+  _animate: boolean | undefined = undefined;
+
+  constructor(props: DeckProps) {
+    super();
+
+    this.props = props;
+  }
+
+  onAdd(): this {
+    const pane = this.getPane();
+    if (!pane) {
+      return this;
+    }
+
+    this._container = L.DomUtil.create('div');
+    this._container.className = 'leaflet-layer';
+    if (this.#getZoomAnimated()) {
+      L.DomUtil.addClass(this._container, 'leaflet-zoom-animated');
+    }
+
+    pane.appendChild(this._container);
+    this._deck = createDeckInstance(this._map, this._container, this._deck, this.props);
+    this.#update();
+
+    return this;
+  }
+
+  onRemove(_map: L.Map): this {
+    if (!this._container || !this._deck) {
+      return this;
+    }
+
+    L.DomUtil.remove(this._container);
+    this._container = undefined;
+
+    this._deck.finalize();
+    this._deck = undefined;
+
+    return this;
+  }
+
+  getEvents(): {[name: string]: L.LeafletEventHandlerFn} {
+    return {
+      viewreset: this.#reset.bind(this),
+      movestart: this.#onMoveStart.bind(this),
+      moveend: this.#onMoveEnd.bind(this),
+      zoomstart: this.#onZoomStart.bind(this),
+      zoom: this.#onZoom.bind(this),
+      zoomend: this.#onZoomEnd.bind(this),
+      ...(this.#getZoomAnimated()
+        ? {zoomanim: this.#onAnimZoom.bind(this) as L.LeafletEventHandlerFn}
+        : {})
+    };
+  }
+
+  setProps(props: DeckProps): void {
+    Object.assign(this.props, props);
+
+    if (this._deck) {
+      this._deck.setProps(props);
+    }
+  }
+
+  pickObject(opts: Parameters<Deck['pickObject']>[0]): ReturnType<Deck['pickObject']> {
+    if (!this._deck) {
+      return null;
+    }
+
+    return this._deck.pickObject(opts);
+  }
+
+  pickMultipleObjects(
+    opts: Parameters<Deck['pickMultipleObjects']>[0]
+  ): ReturnType<Deck['pickMultipleObjects']> {
+    if (!this._deck) {
+      return [];
+    }
+
+    return this._deck.pickMultipleObjects(opts);
+  }
+
+  pickObjects(opts: Parameters<Deck['pickObjects']>[0]): ReturnType<Deck['pickObjects']> {
+    if (!this._deck) {
+      return [];
+    }
+
+    return this._deck.pickObjects(opts);
+  }
+
+  #getMap(): L.Map & {_animatingZoom: boolean; _getMapPanePos: () => L.Point} {
```

Comment:
Nit: are these private declarations? We usually use the typescript private keyword and a leading underscore